### PR TITLE
[ActionSheet] Graduate theming extension to ready

### LINKED ISSUE #7160: [ActionSheet] Mark ActionSheet theming extension ready
This was filed as an internal issue. If you are a Googler, please visit [b/130716161](http://b/130716161) for more details.

---
#### Internal data
- Associated internal bug: [b/130716161](http://b/130716161)

### PR DESCRIPTION
**Pull request by codeman7**
## Context
As part of our work to move away from _themer_ objects and onto _theming extension_ we need to graduate all of our components theming extensions to the MaterialComponents pod. 

## The fix
This PR updates the ActionSheet theming extension to the MaterialComponents pod, as well as updating examples and unit test. Additionally this removes the documentation for _themers_ since they were never out of the MaterialComponentsBeta podspec. The documentation for _themers_ is replaced with documentation for the _theming extension_.

## Bug
Closes #7160

### CODE REVIEW

File: components/ActionSheet/tests/unit/Theming/MDCActionSheetThemingTest.m [hunk 1]
```diff
@@ -63,10 +67,17 @@ - (void)testActionSheetThemingTest {
   // Given
   MDCTypographyScheme *typographyScheme =
       [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
+  MDCActionSheetAction *fakeActionOne = [MDCActionSheetAction actionWithTitle:@"Action 1"
```
Comment (romoore) [inline components/ActionSheet/tests/unit/Theming/MDCActionSheetThemingTest.m]: Looks like some functional changes got mixed-in with the Beta-to-Ready changes. Could you please revert these or split them into a separate PR?

**Review by romoore — DISMISSED** [review]

File: components/ActionSheet/tests/unit/Theming/MDCActionSheetThemingTest.m (hunk 1 above)
Comment (codeman7) [reply components/ActionSheet/tests/unit/Theming/MDCActionSheetThemingTest.m]: This is addressed in #7183 which I will merge before this lands.

**Review by jverkoey — DISMISSED** [review]
Please update the README as well.

File: components/ActionSheet/README.md [hunk 2]
```diff
@@ -51,7 +49,7 @@ the screen and displays actions a user can take.
 
 `MDCActionSheetController` is a material design implementation of UIAlertControllerStyleActionSheet.
 
-Action Sheet is currently an [beta component](../../contributing/beta_components.md). Therefore, clients that
+Action Sheet is currently a [beta component](docs/../../contributing/beta_components.md). Therefore, clients that
```
Comment (jverkoey) [inline components/ActionSheet/README.md:56]: This statement looks incorrect; Action Sheet is no longer a beta component.

File: components/ActionSheet/docs/color-theming.md [hunk 3]
```diff
@@ -1,39 +0,0 @@
-### Color Theming
```
Comment (jverkoey) [inline components/ActionSheet/docs/color-theming.md:1]: Please keep this original documentation until we've deleted the themers. Annotate these docs that they are representing a to-be-deleted set of APIs though so that their state is clear.

File: components/ActionSheet/docs/typography-theming.md [hunk 4]
```diff
@@ -1,39 +0,0 @@
-### Typography Theming
```
Comment (jverkoey) [inline components/ActionSheet/docs/typography-theming.md:1]: Similarly here; please keep this around and annotate it as to-be-deleted.

File: components/ActionSheet/README.md [hunk 5]
```diff
@@ -37,8 +37,6 @@ the screen and displays actions a user can take.
   - [Typical use](#typical-use)
 - [MDCActionSheetController vs. UIAlertControllerStyleActionSheet](#mdcactionsheetcontroller-vs.-uialertcontrollerstyleactionsheet)
 - [Extensions](#extensions)
```
Comment (jverkoey) [inline components/ActionSheet/README.md:39]: I think you may need to add the theming doc to this table of contents and re-run the ActionSheet docs generator.

**Review by jverkoey — DISMISSED** [review]
A few fairly significant change requests.

File: components/ActionSheet/README.md (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/README.md:56]: Done

File: components/ActionSheet/docs/color-theming.md (hunk 3 above)
Comment (codeman7) [reply components/ActionSheet/docs/color-theming.md:1]: Done

File: components/ActionSheet/docs/typography-theming.md (hunk 4 above)
Comment (codeman7) [reply components/ActionSheet/docs/typography-theming.md:1]: Done

File: components/ActionSheet/README.md (hunk 5 above)
Comment (codeman7) [reply components/ActionSheet/README.md:39]: Done

File: components/ActionSheet/docs/theming.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+### Theming
+
+You can theme an MDCActionSheet to match the Material Design style by using a theming extension. The content below assumes you have read the article on [Theming](../../docs/theming.md).
```
Comment (jverkoey) [inline components/ActionSheet/docs/theming.md]: I think this needs one more level of unnesting.

**Review by jverkoey — APPROVED** [review]

File: components/ActionSheet/docs/theming.md (hunk 6 above)
Comment (codeman7) [reply components/ActionSheet/docs/theming.md]: Done

**Review by jverkoey — APPROVED** [review]

File: components/ActionSheet/README.md [hunk 7]
```diff
@@ -37,8 +37,10 @@ the screen and displays actions a user can take.
...
-  - [Color Theming](#color-theming)
-  - [Typography Theming](#typography-theming)
+  - [Theming](#theming)
+  - [How to theme an MDCActionSheet](#how-to-theme-an-mdcactionsheet)
+  - [Color Theming (To be deleted)](#color-theming-(to-be-deleted))
+  - [Typography Theming (To be deleted)](#typography-theming-(to-be-deleted))
```
Comment (romoore) [inline components/ActionSheet/README.md:43]: Looks like this link needs to be fixed as well. (https://github.com/material-components/material-components-ios/issues/7068)
